api/v1alpha1/dspipeline_types.go

```diff
@@ -381,9 +381,23 @@ type SecretKeyValue struct {
 }
 
 type DSPAStatus struct {
+	// +kubebuilder:validation:Optional
+	Components ComponentStatus    `json:"components,omitempty"`
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+type ComponentStatus struct {
+	// +kubebuilder:validation:Optional
+	Envoy     ComponentDetailStatus `json:"envoy,omitempty"`
```

Comment:
(I'm going to comment here, but the change would apply to everywhere ... it would be a big change.)

I don't like using the name `Envoy` here and elsewhere in this PR, for 2 reasons. 

1 What this is actually capturing is the readiness status of the *MLMD proxy*. So it should have "MLMD" in the name. (We could add an Envoy proxy elsewhere someday, and then we'd be in trouble.)

2 Today we happen to implement this proxy with Envoy, but the API shouldn't expose that fact where it doesn't need to. Who knows, this could change to not use Envoy down the road.


```suggestion
	MLMDProxy     ComponentDetailStatus `json:"mlmdProxy,omitempty"`
```